src/core/utils.ts

```diff
@@ -187,6 +191,32 @@ export function hasVisibleLegendItems(options: Highcharts.Options) {
   });
 }
 
+export function hasVisibleSecondaryLegendItems(options: Highcharts.Options) {
+  // Note: While native Highcharts supports multiple legend groups, this
+  // implementation simplifies to at most 2 legend groups:
+  // - One group for all axes with opposite=false (primary axes)
```

Comment:
I think it is better to state that only two axes are supported, where one must be primary and the other secondary. Although the code can technically handle both - using more than two axes or two axes that are both primary or both secondary will result in unexpected UX and a11y issues.